OpenOversight/app/utils.py

```diff
@@ -425,31 +429,45 @@ def create_description(self, form):
         date_updated=datetime.datetime.now())
 
 
-def get_uploaded_cropped_image(original_image, crop_data):
+def get_uploaded_image(image, crop_data=None, department_id=None):
```

Comment:
I think the `get_uploaded_image()` function is a bit confusing as it is, mostly because it attempts to accomplish too many tasks and operates differently depending on which view function it was called from. I would suggest refactoring it out into two different functions:

`upload_image_to_s3(department_id, image_data<io.BytesIO>):`
1. hash `image_data`. if hash found in DB, return existing Image
2. determine image format using `imghdr.what(h=image_data)`; only continue if format recognized by imghdr
3. determine date taken if possible
4. upload to S3 and return new Image instance (don't set `is_tagged`)

`crop_image(image<Image>, crop_data=None):`
1. read in raw image data via `image.filepath` (url or local file) and create a BytesIO buffer from it
2. determine image type using imghdr again, reject if unrecognized
3. crop image: `cropped_image = Pimage.open(image_data).crop(crop_data)` (or `.thumbnail((300,300))` if `crop_data==None`)
4. Save the cropped Pimage into a new BytesIO buffer via `Pimage.save()` using the image format determined from imghdr
5. `return upload_image_to_s3(department_id, cropped_image_buffer)`

A couple things to note: 1) Neither function sets the `is_tagged` field on the resulting Image; that's the responsibility of the caller/view function. 2) The uploaded image (original or cropped) is never saved to disk during this process; it's passed around as a BytesIO stream. That's not functionally necessary, but it does provide added security as it precludes a local file inclusion attack.